src/main/java/org/opensearch/neuralsearch/processor/chunker/FixedTokenLengthChunker.java

```diff
@@ -35,128 +25,156 @@ public class FixedTokenLengthChunker implements Chunker {
         }
+        this.tokenizer = validateStringParameters(parameters, TOKENIZER_FIELD, DEFAULT_TOKENIZER, false);
+        this.tokenConcatenator = validateStringParameters(parameters, TOKEN_CONCATENATOR_FIELD, DEFAULT_TOKEN_CONCATENATOR, true);
+    }
 
-        if (parameters.containsKey(TOKENIZER_FIELD)) {
-            if (!(parameters.get(TOKENIZER_FIELD) instanceof String)) {
-                throw new IllegalArgumentException(
-                    "fixed length parameter [" + TOKENIZER_FIELD + "] cannot be cast to [" + String.class.getName() + "]"
-                );
-            }
-            this.tokenizer = parameters.get(TOKENIZER_FIELD).toString();
+    public static String validateStringParameters(
+        Map<String, Object> parameters,
+        String fieldName,
+        String defaultValue,
+        boolean allowEmpty
+    ) {
+        if (!parameters.containsKey(fieldName)) {
+            // all parameters are optional
+            return defaultValue;
+        }
+        if (!(parameters.get(fieldName) instanceof String)) {
+            throw new IllegalArgumentException("Chunker parameter [" + fieldName + "] cannot be cast to [" + String.class.getName() + "]");
+        } else if (StringUtils.isEmpty(parameters.get(fieldName).toString()) && !allowEmpty) {
+            throw new IllegalArgumentException("Chunker parameter: " + fieldName + " should not be empty.");
         }
+        return (String) parameters.get(fieldName);
+    }
+
+    private int validatePositiveIntegerParameter(Map<String, Object> parameters, String fieldName, int defaultValue) {
+        // this method validate that parameter is a positive integer
+        if (!parameters.containsKey(fieldName)) {
+            // all parameters are optional
+            return defaultValue;
+        }
+        String fieldValue = parameters.get(fieldName).toString();
+        if (!(NumberUtils.isParsable(fieldValue))) {
+            throw new IllegalArgumentException(
+                "fixed length parameter [" + fieldName + "] cannot be cast to [" + Number.class.getName() + "]"
+            );
+        }
+        if (NumberUtils.createInteger(fieldValue) <= 0) {
+            throw new IllegalArgumentException("fixed length parameter [" + fieldName + "] must be positive");
+        }
+        return Integer.valueOf(fieldValue);
     }
 
     /**
      * Return the chunked passages for fixed token length algorithm
      *
      * @param content input string
+     * @param parameters a map containing parameters, containing the following parameters
+     * 1. tokenizer the <a href="https://opensearch.org/docs/latest/analyzers/tokenizers/index/">analyzer tokenizer</a> in OpenSearch
+     * 2. token_limit the token limit for each chunked passage
+     * 3. overlap_rate the overlapping degree for each chunked passage, indicating how many token comes from the previous passage
+     * 4. max_token_count the max token limit for the tokenizer
```

Comment:
For this java doc, please update parameters with runtimeParameters (max token count).